fix: Added Pitstops, Qualifying, Circuits, Constructors, Drivers Standings and Constructors standings endpoint docs (#83)
jolpica/jolpica-f1: docs/endpoints/drivers.md

[← Documentation Home](/docs/README.md)
# Drivers

> Gets list of drivers in alphabetical order by driverId

Reviewer: ```suggestion
Returns a list of drivers in alphabetical order by `driverId`.
```

**URL** : `/ergast/f1/drivers/`

[Available Query Parameters](/docs/README.md#query-parameters)

---

## Route Parameters:

### Season

> **Filters only drivers that participated in a specified season. Year numbers are valid as is `current` to get the current season list of drivers**

Reviewer: ```suggestion
Filters only drivers that participated in a specified season. Year numbers are valid as is `current` to get the current season list of drivers.
```

`/{season}/` -> ex: `/ergast/f1/2024/drivers/`

**Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`

---

### Round

> **Filters only drivers that participated in a specified round of a specific season. Round numbers 1 -> n races are valid as well as `last`**

Reviewer: ```suggestion
Filters only drivers that participated in a specified round of a specific season. Round numbers 1 -> `n` races are valid as well as `last`.
```

`/{round}/` -> ex: `/ergast/f1/2024/1/drivers/`

> **Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first route after `/ergast/f1/{season}`

Reviewer: ```suggestion
**Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first argument after `/ergast/f1/{season}`.
```

---

### circuits

> **Filters for only drivers who have participated in a race at a given circuit**

Reviewer: ```suggestion
Filters for only drivers who have participated in a race at a given circuit.
```

`/circuits/{circuitId}/` -> ex: `/ergast/f1/2024/circuits/albert_park/drivers/`

---

### constructors

**Filters for only drivers who have raced for a specified constructor**

`/constructors/{constructorId}/` -> ex: `/ergast/f1/constructors/williams/drivers/`

---

### drivers

> **Filters for only drivers that match the specific driverId**

Reviewer: ```suggestion
Filters for only drivers that match the specific `driverId`.
```

> **Filters for only drivers that finished a race with a lap that was the ranked in the specified position**

Reviewer: ```suggestion
Filters for only drivers that finished a race with a lap that was the ranked in the specified position.
```

> **Filters for only drivers who has started a race in a specific grid position**

Reviewer: ```suggestion
Filters for only drivers who have started a race in a specific grid position.
```

`/grid/{gridPosition}/` -> ex: `/ergast/f1/grid/1/drivers/`

---

### results

> **Filters for only drivers who has finished a race in a specific position**

Reviewer: ```suggestion
Filters for only drivers who have finished a race in a specific position.
```

`/results/{finishPosition}/` -> ex: `/ergast/f1/results/1/drivers/`

---

### status

> **Filters for only drivers who have finished a race with a specific statusId**

Reviewer: ```suggestion
Filters for only drivers who have finished a race with a specific `statusId`.
```

`MRData.DriverTable.Drivers` : The list of all drivers returned.

`MRData.DriverTable.Drivers[i]` : A given driver object.

---

## Driver Object Fields:

|Field|Always Included|Description|type
|---|:---:|---|---|
|driverId|✅|Unique ID of the Driver|String
|permanentNumber|🟡|Permanent Number assigned to the driver|String
|code|🟡|Driver Code, usually 3 characters|String

> |url|✅|Wikipedia URL to the Drivers profile|String
> |givenName|✅|First name|String
> |familyName|✅|Last name|String
> |dateOfBirth|✅|Date of Birth (YYYY-MM-DD format)|String
> |nationality|✅|Nationality of Driver|String

Reviewer 2: ```suggestion
|url| 🟡 |Wikipedia URL to the Drivers profile|String
|givenName|✅|First name|String
|familyName|✅|Last name|String
|dateOfBirth| 🟡 |Date of Birth (YYYY-MM-DD format)|String
|nationality| 🟡 |Nationality of Driver|String
```
